### ISSUE TITLE
Disable it outside of the US

### ISSUE THREAD

**Issue by martixy** [fightforthefuture/battleforthenet-widget#145]
Might wanna disable the popup outside the US. People who are not US citizens or living in the US can't do shit, so it's just an annoying popup to the _majority_ of the world.

**Comment by dob9601** [comment]
Read other issues before submitting one yourself. https://github.com/fightforthefuture/battleforthenet-widget/issues/144 The removal of net neutrality will impact upon people outside of the US too

**Comment by rubbingalcoholic** [comment]
People who are in the U.S. but use foreign VPNs should see the pop-up.
Geofencing doesn't work. The Internet is global. Net neutrality affects
everyone lol

**Comment by SG0505** [comment]
Yes, but asking people who live outside the US to contact their member of congress seems a bit pointless.
I'm a Brit who lives in London.   To which member of congress am I being asked to write?

**Comment by rubbingalcoholic** [comment]
I agree that we need a better call to action describing the problem from an
international standpoint, ie. how stupid policy in the U.S. will affect the
world wide web, and perhaps provide action alternatives.

In your case, perhaps the SaveTheInternet coalition in the U.K. could
pressure your members of parliament to ask the U.S. Congress to intervene
and stop the F.C.C.

If anything, the fact that you are inconvenienced by this shows how serious
the problem is, when bad U.S. policy starts to put the sites you use out of
business. Sorry I don't have a better solution in this case, but I welcome
any creative ideas you can come up with.

On Dec 13, 2017 4:28 PM, "SG0505" <notifications@github.com> wrote:

Yes, but asking people who live outside the US to contact their member of
congress seems a bit pointless.
I'm a Brit who lives in London. To which member of congress am I being
asked to write?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/fightforthefuture/battleforthenet-widget/issues/145#issuecomment-351545741>,
or mute the thread

**Comment by SG0505** [comment]
Personally I'm not inconvenienced by it.    It's just that, for someone who is concerned about net neutrality, if only for how it's going to affect my friends in the USA, it's a bit irritating to be asked to do something that's completely impossible.

I guess I could write to my Member of Parliament about it, but since I'm not at all clear about how the US decision is likely to affect me directly or indirectly, so I'm not really sure what I should say to him.     I know what the possible ill effects of abandoning net neutrality in the USA will be for  customers of US ISPs but it's less clear to me what they'll be for Brits.    Quite possibly there will be all sorts of consequences outside the USA, but I just don't know what they might be.

**Comment by RookyGhost** [comment]
well, considering that the internet is a two-way street, the most basic way in which it'll affect us non-us citizens is preventing us from contacting people/sites/services within the US they way we used to do.
it would be wise perhaps to include a cookie?

**Comment by Julian2000nl** [comment]
I agree with this. Most of us Europeans don't care about the problems of the United States. We have a law passing on allowing mass surveillance here in The Netherlands and we don't give everyone around the world popups.

Going to look into this today and will see if I can contribute some code to make this happen.

**Comment by rubbingalcoholic** [comment]
@Julian,

Please don't waste your time adding geofencing. I could trivially add that
to this repo if I or my colleagues thought it was a good idea. But again
there are issues:

- We encourage people to use VPNs, which tend to break geofencing
- Many of our webmasters are privacy conscious, and doing a location lookup
(even first party) can be seen as hostile
- Geolookup will slow things down, and it's frankly not webmaster-friendly.
They chose to embed this on their sites, not the other way around.
- We would much rather see one or more international alternative actions a
user can take, by clicking a "Not in the U.S.?" link.

If you want to contribute, please focus your efforts on that last part.
That's the only reason I'm keeping this issue open, though it should be
renamed.

**Comment by SG0505** [comment]
I'd be interested to know what further action I can take, since I'm outside the US, in the UK.   

I'm perfectly prepared to write to my Member of Parliament, for example, if someone can provide me with some good reasons to give him explaining why a foreign legislator should be contacting the US FCC about a matter that he will probably consider, at first sight, to be a purely US domestic matter.

**Comment by kenmickles** [comment]
I think a "not in the US?" page with a list of international actions would be a great addition.

**Comment by hishnash** [comment]
@rubbingalcoholic

I believe you are forgetting not everyones english is perfect. If your english is not good and you come across a page like this mentioning the US this might well look like the US gov is blocking the page etc 

the first time I saw this, with a russian IP even through I am a native brit I defaulted to thinking that the page was blocked since I was coming from russia (yes this is somewhat common)

**Comment by gingerbeardman** [comment]
Please

**Comment by N0ps32** [comment]
> "We need one more vote to win in the Senate. Write and call now! "  

Great and how am I supposed to do that in Europe?  
Americans should really stop assuming they are the center of the Universe.  

Please disable the Popup for people that don't live in the US. It's a waste of time, waste of KB that my modem has to transfer and completely irrelevant.

Maybe I'll contribute a filter to uBlock origin to just completely block all battleforthenet widgets. Anyone interested? Otherwise I'll just block it locally.

**Comment by gingerbeardman** [comment]
@N0ps32 a secondary component will also be needed to reinstate page scrolling (CSS/JS)

**Comment by milko1** [comment]
At least allow change of language, either set in the widget or take brower’s, with a default (to EN) when not available.

**Comment by milko1** [comment]
created this fork & PR to address this issue (& more), at least allowing web admins to select countries this would show for https://github.com/fightforthefuture/battleforthenet-widget/pull/160
